### PR TITLE
Adds short people, body size options port

### PR DESCRIPTION
**Pull request by ElorgRHG**
## About The Pull Request
Port of https://github.com/BeeStation/BeeStation-Hornet/pull/6046
As the original PR described it:
Short/Normal/Tall
All species (except IPCs) can now choose between "Normal" and "Short" height. Short height scales their sprite by 0.93 and Tall 1.03, which I hand-picked to be the best looking scaling.

Ever wanted to play as an even shorter (or taller) Kepori? Now you can!
image

... works on other species too I guess...
image

## Why It's Good For The Game
Now you can project your real-life insecurities by making a characters that is _slightly_ taller (or shorter) than all the other ones... or well, it looks nice I suppose...

## Changelog
:cl:Kapu1178
add: You can now pick body size for characters: short, normal or tall.
/:cl:

### CODE REVIEW

**Review by Kapu1178 — APPROVED** [review]

**Comment by Martinpachu** [comment]
why not ipcs :( i want a tiny robot too

**Comment by ZephyrTFA** [comment]
Because this touches body code it'll probably need to be test merged

File: code/datums/dna.dm [hunk 1]
```diff
@@ -290,6 +292,20 @@
...
+/datum/dna/proc/update_body_size()
+	if(!holder)
+		return
+	var/static/list/feat2num = list("Normal" = BODY_SIZE_NORMAL, "Tall" = BODY_SIZE_TALL, "Short" = BODY_SIZE_SHORT)
+	var/desired_size = feat2num[features["body_size"]]
+
+	if(desired_size == current_body_size)
+		return
+
+	var/change_multiplier = desired_size / current_body_size
```
Comment (francinum) [inline code/datums/dna.dm:306]: `current_body_size` can be null.
```
proc name: update body size (/datum/dna/proc/update_body_size)
usr: Piperdoots/(Piperdoots)
usr.loc: (start area (8, 174, 1))
src: /datum/dna (/datum/dna)
call stack:
/datum/dna (/datum/dna): update body size()
/datum/preferences (/datum/preferences): copy to(Melari (/mob/living/carbon/human/dummy), 1, 1, 1, 0, 1)
/datum/preferences (/datum/preferences): update preview icon(1, 1)
/datum/preferences (/datum/preferences): ShowChoices(Piperdoots (/mob/dead/new_player))
Piperdoots (/mob/dead/new_player): Topic("src=\[mob_69];show_preferences...", /list (/list))
Piperdoots (/client): Topic("src=\[mob_69];show_preferences...", /list (/list), Piperdoots (/mob/dead/new_player))
Piperdoots (/client): Topic("src=\[mob_69];show_preferences...", /list (/list), Piperdoots (/mob/dead/new_player))
Piperdoots (/client): Topic("src=\[mob_69];show_preferences...", /list (/list), Piperdoots (/mob/dead/new_player))
```

Comment (francinum) [reply code/datums/dna.dm:306]: Ping me on discord when this issue is resolved.

Comment (francinum) [reply code/datums/dna.dm:306]: This appears to be a result of a race condition with preference code, where `prefs.features` is not properly initialized and thus is missing the body_size selection.

**Review by MarkSuckerberg — DISMISSED** [review]
Doing this to block merging, see Fran's requests above

**Comment by WindowsErrors** [comment]
5' 6" crew holds the entire Caribbean Sector hostage with a single pirate ship

**Comment by Ms-Mee** [comment]
Is there any specific reason for the features list to not normally include body_size?

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -83,7 +83,33 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+							"spider_spinneret" = "Plain",
+							"spider_mandibles" = "Plain",
+							"squid_face" = "Squidward",
+							"ipc_screen" = "Blue",
+							"ipc_antenna" = "None",
+							"ipc_chassis" = "Morpheus Cyberkinetics(Greyscale)",
+							"kepori_feathers" = "Plain",
+							"kepori_body_feathers" = "Plain",
+							"flavor_text" = ""
+						)
```
Comment (Kapu1178) [inline code/modules/client/preferences.dm:112]:
```suggestion
							"flavor_text" = ""
							"body_size = "Normal"
						)
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences.dm (hunk 2 above)
Comment (Kapu1178) [inline code/modules/client/preferences.dm]:
```suggestion
							"flavor_text" = "",
							"body_size" = "Normal"
						)
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
fuckin github

**Review by Kapu1178 — APPROVED** [review]

**Comment by Sylphily** [comment]
Is there a reason why this hasn't been merged yet?

**Comment by Kapu1178** [comment]
> Is there a reason why this hasn't been merged yet?

Review requested for @MarkSuckerberg

**Review by MarkSuckerberg — APPROVED** [review]

**Pull request merged**